File: solutions/dart/pig-latin/1/lib/pig_latin.dart
String translate(String phrase) {
  final vowelPattern = RegExp(r'^(xr|yt|[aeiou])');
  final consonantPattern = RegExp(r'^([^aeiou]?qu|[^aeiou]+)([aeiouy].*)$');
  final words = phrase.split(' ');

  return words.map((word) {
    if (vowelPattern.hasMatch(word)) {
      return word + 'ay';
    } else if (consonantPattern.hasMatch(word)) {
      final match = consonantPattern.firstMatch(word);
      return '${match?.group(2)}${match?.group(1)}ay';
    }
    return word; // Fallback for unmatched patterns, though all cases should be covered.
  }).join(' ');
}

File: solutions/python/pig-latin/1/pig_latin.py
"""
I asked AI to translate my previoud Dart code into Python. 
"""
import re

def translate(text):
    words=text.split(" ")

    translated_words=[]
    for word in words:
        if re.match(r"^(a|e|i|o|u|xr|yt)", word):
            translated_word=word+'ay'
        elif re.match(r"^([^aeiou]*qu)(.*)", word):
            match=re.search(r"([^aeiou]*qu)(.*)", word)
            translated_word=f"{match.group(2)}{match.group(1)}ay"
        elif re.match(r"^([^aeiouy]+)(y.*)", word) or (len(word) == 2 and word.startswith("y")):
            match=re.search(r"^([^aeiouy]*)(y.*)", word)
            translated_word=f"{match.group(2)}{match.group(1)}ay"
        else:
            match = re.search(r"^([^aeiou]+)(.*)", word)
            translated_word = f"{match.group(2)}{match.group(1)}ay"
        
        translated_words.append(translated_word)

    return " ".join(translated_words)



        

